src/repr/scalar/mod.rs

```diff
@@ -802,15 +798,110 @@ impl Interval {
         let ns = f64::from(self.duration.subsec_nanos()) / 1e9;
         s + ns
     }
-}
 
-impl From<SqlInterval> for Interval {
-    fn from(other: SqlInterval) -> Interval {
-        Interval {
-            months: other.months,
-            duration: other.duration,
-            is_positive_dur: other.is_positive_dur,
+    /// Truncate the "head" of the interval, removing all time units greater than `f`.
+    pub fn truncate_high_fields(&mut self, f: DateTimeField) {
+        use std::time::Duration;
+        use DateTimeField::*;
+        match f {
+            Year => {}
+            Month => {
+                println!(
+                    "Have {} months; trimming years, should have {}",
+                    self.months,
+                    self.months % 12
+                );
+                self.months %= 12;
+                println!("Now Have {} months", self.months);
+                println!("Self {:?}", self);
+            }
+            Day => {
+                self.months = 0;
+            }
+            hms => {
```

Comment:
another super tiny nit: the combination of enum variants and a variable at the end of this would be slightly more obvious if this didn't `use DateTimeField::*;` at the beginning of this method.

Reply:
Great catch